### ISSUE TITLE
Feature request: Extend functionality of sorted collections.

### ISSUE THREAD

**Issue by bent95** [louthy/language-ext#648]
Over time I’ve realized that I really dislike dealing with mutable collections, but there are a few features of the sorted collections in the C5 library in particular that I use a lot, and haven’t found a simple replacement for in this libraries:

* TryPredecessor(item/key): This function returns the closest item/key-value pair in the collection which is less than the argument. If no such item is found it returns nothing. TryWeakPredecessor works identically except it looks for an item which is less than or equal to the argument. And of course TrySuccessor and TryWeakSuccessor are the same thing but the other way around. 

* RangeFromTo/Slice(min, max): Works similarly to the FindRange(keyFrom, keyTo) in Map except it returns a slice of the Map with key-value pairs intact, instead of just the values in the range. Having such a function in Set would be useful as well.

* FindMin()/FindMax(): Returns the item/key-value pair which is sorted lowest or highest respectively.

Also I just want to say that that I am very excited to begin using this library! I’ve been interested in learning functional programming for a while, and it is very appealing to me to do so within the familiarity of C#.

**Comment by bender2k14** [comment]
Can you add links to the documentation of these functions in the C5 library?

**Comment by louthy** [comment]
Seems reasonable.  I'll stick it on the list.

**Comment by bent95** [comment]
> 
> 
> Can you add links to the documentation of these functions in the C5 library?

All of the mentioned functions are documented in the C5 manual from page 91 to 93. The manual can be downloaded here: https://www.itu.dk/research/c5/latest/ITU-TR-2006-76.pdf

**Comment by louthy** [comment]
@bent95 released in `v3.3.29`

## `Map<K, V>` and `Map<OrdK, K, V>`
```c#
    Map<K, V> Slice(K keyFrom, K keyTo);
    Option<(K Key, V Value)> FindPredecessor(K key);
    Option<(K Key, V Value)> FindExactOrPredecessor(K key);
    Option<(K Key, V Value)> FindSuccessor(K key);
    Option<(K Key, V Value)> FindExactOrSuccessor(K key);
    Option<(K Key, V Value)> Max;
    Option<(K Key, V Value)> Min;
```
## `Set<A>` and `Set<OrdA, A>`
```c#
    IEnumerable<A> FindRange(A keyFrom, A keyTo);
    Set<A> Slice(A keyFrom, A keyTo);
    Option<A> FindPredecessor(A key);
    Option<A> FindExactOrPredecessor(A key);
    Option<A> FindSuccessor(A key);
    Option<A> FindExactOrSuccessor(A key);
    Option<A> Max;
    Option<A> Min;
```
